cpp/src/arrow/util/io-util.cc

```diff
@@ -373,8 +419,22 @@ Status FileOpenReadable(const PlatformFilename& file_name, int* fd) {
                            _O_RDONLY | _O_BINARY | _O_NOINHERIT, _SH_DENYNO, _S_IREAD);
   ret = *fd;
 #else
-  ret = *fd = open(file_name.ToNative().c_str(), O_RDONLY | O_BINARY);
```

Comment:
Does O_BINARY actually do anything here?

Reply:
On POSIX it was defined to 0, so no. On Windows it's required to avoid automatic newline translation.